<?xml version="1.0" ?>
<launch>
  <param name="use_sim_time" value="true"/>
  <arg name="path" default="/home/$(optenv USER gsl)/datasets/omni_sim/omni_simulated.bag"/>
  <arg name="debug" default="false"/>
  <arg name="publish" default="true"/>
  <arg name="rate" default="1.0"/>
  <arg name="particles" default="250"/>
  <arg name="performer_path" default="/home/$(optenv USER gsl)/datasets/pfuclt/"/>
  <arg name="performer_file" default="experimentX-y_Nrobots_datasetD_runR.bag"/>
  <node pkg="rosbag" type="play" name="player" required="true" output="screen" args="--quiet --clock --rate=$(arg rate) $(arg path)"/>
  <node name="performer" pkg="pfuclt_omni_dataset" type="pfuclt_omni_dataset" output="screen" args="--debug $(arg debug) --publish $(arg publish)">
    <param name="percentage_to_keep" value="50"/>
    <param name="velocity_estimator_stack_size" value="10"/>
    <param name="predict_model_stddev" value="30"/>
    <param name="particles" value="$(arg particles)"/>
    <param name="OMNI1_alpha" value="0.015, 0.1, 0.5, 0.001"/>
    <param name="OMNI2_alpha" value="0.015, 0.1, 0.5, 0.001"/>
    <param name="OMNI3_alpha" value="0.015, 0.1, 0.5, 0.001"/>
    <remap to="/gtData" from="/gtData_4robotExp"/>
    <rosparam param="PLAYING_ROBOTS">[1, 1, 1]</rosparam>
    <param name="MAX_ROBOTS" value="3"/>
    <param name="ROB_HT" value="0.81"/>
    <param name="MY_ID" value="1"/>
    <param name="NUM_TARGETS" value="1"/>
    <param name="LANDMARK_COV/K1" value="0.02"/>
    <param name="LANDMARK_COV/K2" value="0.005"/>
    <param name="LANDMARK_COV/K3" value="0.002"/>
    <param name="LANDMARK_COV/K4" value="0.005"/>
    <param name="LANDMARK_COV/K5" value="0.005"/>
    <param name="NUM_LANDMARKS" value="10"/>
    <param name="LANDMARKS_CONFIG" value="$(find pfuclt_omni_dataset)/config/landmarks.csv"/>
    <param name="USE_CUSTOM_VALUES" value="false"/>
    <rosparam param="POS_INIT">[6.425334211875039, 0.1777077117708732, -0.0179766297647328, 3.5339684784580716, -1.5548059361599864, 0.26315989227734127, 6.502996727127533, -3.2767528626150257, -1.348320489895425]</rosparam>
    <rosparam param="CUSTOM_PARTICLE_INIT">[6.405334211875039,6.445334211875038,0.1577077117708732,0.19770771177087318,-0.0379766297647328,0.0020233702352672016,3.5139684784580716,3.5539684784580716,-1.5748059361599864,-1.5348059361599864,0.24315989227734128,0.2831598922773413,6.482996727127533,6.522996727127532,-3.2967528626150258,-3.2567528626150257,-1.368320489895425,-1.328320489895425,2.2906480633595594,2.3306480633595594,-4.354732783004476,-4.314732783004477,2.588421809187989,2.628421809187989]</rosparam>
  </node>
  <node pkg="rosbag" name="record_performance" type="record" args="--all --output-name=$(arg performer_path)/$(arg performer_file)"/>
</launch>
